### ISSUE TITLE
Difficult response after request from an TLS client which not supporting TLS SNI 

### ISSUE THREAD

**Issue by wodo** [kennethreitz/httpbin#376]
We did some tests with `openssl s_client -connect 23.23.115.5:443` to simulate our embedded client.
We can't get an connection successfully with this method and got always "Internal_error" . After digging deeper we found  `openssl s_client -connect 23.23.115.5:443 -servername www.httpbin.org` is working fine.

The https://tools.ietf.org/html/rfc6066 states:

> If the server understood the ClientHello extension but does not recognize the server name, the server SHOULD take one of two actions: either abort the handshake by sending a fatal-level unrecognized_name(112) alert or continue the handshake.

It would be very nice to change the response to "unrecognized_name" instead of "Internal_error"
This will help others to find the root cause much faster ;-)

**Comment by sigmavirus24** [comment]
This likely requires @kennethreitz to talk to the right thing at their hosting provider.

**Comment by kennethreitz** [comment]
We require SNI, I believe.

**Comment by sigmavirus24** [comment]
@kennethreitz right, I believe that @wodo is explaining that what is being returned currently is incorrect for that requirement.

**Comment by javabrett** [comment]
This issue should be closed because:

1. _httpbin.org_ requires SNI, which is fairly common and reasonable these days, and
2. The RFC section quoted is not actually being violated.  Note that it says:

> If the server understood the ClientHello extension but does not recognize the server name

This does not apply in the case of `openssl s_client -connect 23.23.115.5:443`, because the client does not send an SNI extension block in this case, therefor it sent no server name, and did not send an unrecognized server name.
